sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveUtilsSuite.scala

```diff
@@ -33,4 +33,13 @@ class HiveUtilsSuite extends QueryTest with SQLTestUtils with TestHiveSingleton
       assert(conf(ConfVars.METASTORE_END_FUNCTION_LISTENERS.varname) === "")
     }
   }
+
+  test("newTemporaryConfiguration respect spark.hadoop.foo=bar in SparkConf") {
+    sys.props.put("spark.hadoop.foo", "bar")
```

Comment:
The test says we should respect hadoop conf in `SparkConf`, but why we handle system properties?